src/transaction.ts

```diff
@@ -161,7 +197,53 @@ class Transaction extends DatastoreRequest {
         : () => {};
     const gaxOptions =
       typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
-    this.#runCommit(gaxOptions, callback);
+    const resolver: Resolver<google.datastore.v1.ICommitResponse> = resolve => {
+      this.#runCommit(
+        gaxOptions,
+        (err?: Error | null, resp?: google.datastore.v1.ICommitResponse) => {
+          resolve({err, resp});
+        }
+      );
+    };
+    this.#withBeginTransaction(gaxOptions, resolver).then(
+      (response: UserCallbackData<google.datastore.v1.ICommitResponse>) => {
+        callback(response.err, response.resp);
+      }
+    );
+  }
+
+  /**
+   * If the transaction has not begun yet then this function ensures the transaction
+   * has started before running the resolver provided. The resolver is a function with one
```

Comment:
There is a bigger discussion to be had here. First, before we add a decorator, let's make the changes from https://github.com/googleapis/nodejs-datastore/pull/1206 so that it is easier to identify the right pattern. I see some ways the decorator can be added, but they add too many levels of indirection. I'll think about it some more though. In the mean time I think the changes from https://github.com/googleapis/nodejs-datastore/pull/1206 should be added.

Reply:
That's fine if decorators don't fit, just wondering if it was considered